src/Microsoft.Azure.WebJobs.ServiceBus/Triggers/ServiceBusTriggerBinding.cs

```diff
@@ -122,6 +127,75 @@ public Task<IListener> CreateListenerAsync(ListenerFactoryContext context)
             return factory.CreateAsync(context.CancellationToken);
         }
 
+        internal static IReadOnlyDictionary<string, Type> CreateBindingDataContract(IReadOnlyDictionary<string, Type> argumentBindingContract)
+        {
+            var contract = new Dictionary<string, Type>(StringComparer.OrdinalIgnoreCase);
+            contract.Add("DeliveryCount", typeof(int));
+            contract.Add("DeadLetterSource", typeof(string));
+            contract.Add("ExpiresAtUtc", typeof(DateTime));
+            contract.Add("EnqueuedTimeUtc", typeof(DateTime));
+            contract.Add("MessageId", typeof(string));
+            contract.Add("ContentType", typeof(string));
+            contract.Add("ReplyTo", typeof(string));
+            contract.Add("SequenceNumber", typeof(long));
+            contract.Add("To", typeof(string));
+            contract.Add("Label", typeof(string));
+            contract.Add("CorrelationId", typeof(string));
+
+            if (argumentBindingContract != null)
+            {
+                foreach (KeyValuePair<string, Type> item in argumentBindingContract)
+                {
+                    // In case of conflict, binding data from the value type overrides the built-in binding data above.
+                    contract[item.Key] = item.Value;
+                }
+            }
+
+            return contract;
+        }
+
+        internal static IReadOnlyDictionary<string, object> CreateBindingData(BrokeredMessage value,
+            IReadOnlyDictionary<string, object> bindingDataFromValueType)
+        {
+            var bindingData = new Dictionary<string, object>(StringComparer.OrdinalIgnoreCase);
+
+            SafeAddValue(() => bindingData.Add(nameof(value.DeliveryCount), value.DeliveryCount));
```

Comment:
BrokeredMessage is not very unit testable. Getters on local message properties will throw in many cases, because the message hasn't actually been sent/receive. E.g properties like ExpiresAt, EnqueuedTime, etc. In a full e2e test however, none of these throw.